source/include/signaling_data_types.h

```diff
@@ -0,0 +1,381 @@
+ * @brief Return code of Signaling component.
+ */
+typedef enum SignalingResult
+{
+    SIGNALING_RESULT_OK,
+    SIGNALING_RESULT_BAD_PARAM,
+    SIGNALING_RESULT_SNPRINTF_ERROR,
+    SIGNALING_RESULT_OUT_OF_MEMORY,
+    SIGNALING_RESULT_INVALID_JSON,
+    SIGNALING_RESULT_UNEXPECTED_RESPONSE,
+    SIGNALING_RESULT_INVALID_TTL,
+    SIGNALING_RESULT_INVALID_ENDPOINT,
+    SIGNALING_RESULT_INVALID_PROTOCOL,
+    SIGNALING_RESULT_INVALID_CHANNEL_NAME,
+    SIGNALING_RESULT_INVALID_CHANNEL_TYPE,
+    SIGNALING_RESULT_INVALID_ICE_SERVER_COUNT,
+    SIGNALING_RESULT_INVALID_ICE_SERVER_URIS_COUNT,
+    SIGNALING_RESULT_INVALID_STATUS_RESPONSE,
+    SIGNALING_RESULT_REGION_LENGTH_TOO_LARGE,
+} SignalingResult_t;
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief Protocol type of endpoints while parsing get signaling channel endpoint response.
+ *        Refer to https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_GetSignalingChannelEndpoint.html for more detail.
+ */
+typedef enum SignalingProtocol
+{
+    SIGNALING_PROTOCOL_NONE = 0,
+    SIGNALING_PROTOCOL_WEBSOCKET_SECURE = 1,
+    SIGNALING_PROTOCOL_HTTPS = 2,
+    SIGNALING_PROTOCOL_WEBRTC = 4,
+    SIGNALING_PROTOCOL_MAX = 0xFF,
+} SignalingProtocol_t;
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief Type of signaling channel.
+ *        Refer to https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_ChannelInfo.html#KinesisVideo-Type-ChannelInfo-ChannelType for more detail.
+ */
+/* We cannot name it SignalingChannelType_t in order to avoid naming conflict
+ * with a data type in the existing KVS SDK. */
+typedef enum SignalingTypeChannel
+{
+    SIGNALING_TYPE_CHANNEL_UNKNOWN,
+    SIGNALING_TYPE_CHANNEL_SINGLE_MASTER,
+} SignalingTypeChannel_t;
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief Role of current signaling request.
+ *        Refer to https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-how-it-works.html for more detail.
+ */
+typedef enum SignalingRole
+{
+    SIGNALING_ROLE_NONE = 0,
+    SIGNALING_ROLE_MASTER,
+    SIGNALING_ROLE_VIEWER,
+} SignalingRole_t;
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief Type of signaling message.
+ *        Refer to https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-websocket-apis-7.html for more detail.
+ */
+/* We cannot name it SignalingMessageType_t in order to avoid naming conflict
+ * with a data type in the existing KVS SDK. */
+typedef enum SignalingTypeMessage
+{
+    SIGNALING_TYPE_MESSAGE_UNKNOWN = 0,
+    SIGNALING_TYPE_MESSAGE_SDP_OFFER,
+    SIGNALING_TYPE_MESSAGE_SDP_ANSWER,
+    SIGNALING_TYPE_MESSAGE_ICE_CANDIDATE,
+    SIGNALING_TYPE_MESSAGE_GO_AWAY,
+    SIGNALING_TYPE_MESSAGE_RECONNECT_ICE_SERVER,
+    SIGNALING_TYPE_MESSAGE_STATUS_RESPONSE,
+} SignalingTypeMessage_t;
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief Basic format of the signaling request.

```

Comment:
could you add details on the structure attributes? how about port or protocol?

Reply:
We plan to do documentation for all the libraries later before we go GA.